frontend/src/queries/nodes/DataTable/dataTableLogic.ts

```diff
@@ -61,10 +62,13 @@ export const dataTableLogic = kea<dataTableLogicType>([
     })),
     selectors({
         sourceKind: [(_, p) => [p.query], (query): NodeKind | null => query.source?.kind],
+        sourceFeatures: [(_, p) => [p.query], (query): Set<QueryFeature> => getQueryFeatures(query.source)],
         orderBy: [
-            (_, p) => [p.query],
-            (query): string[] | null =>
-                isEventsQuery(query.source) ? query.source.orderBy || ['timestamp DESC'] : null,
+            (s, p) => [p.query, s.sourceFeatures],
+            (query, sourceFeatures): string[] | null =>
+                sourceFeatures.has(QueryFeature.selectAndOrderByColumns)
+                    ? (query.source as EventsQuery).orderBy || (isEventsQuery(query.source) ? ['timestamp DESC'] : null)
```

Comment:
Can we simplify this double ternary? I had to stare at it pretty hard to figure out what it was doing 😵‍💫